go/vt/vtgate/planbuilder/testdata/from_cases.txt

```diff
@@ -2079,6 +2079,43 @@
   }
 }
 
+# join with USING construct
+"select user.id from user join user_extra using(id)"
+{
```

Comment:
add a test for non-vindex column in projection to see if that works

Reply:
we're not touching this logic at all. all we are doing is rewriting USING to ON in the rewriter